src/collection-preferences/index.tsx

```diff
@@ -126,6 +127,19 @@ export default function CollectionPreferences({
   }
 
   const referrerId = useUniqueId();
+  const tableComponentContext = useTableComponentsContext();
+  if (tableComponentContext?.preferencesRef?.current) {
+    tableComponentContext.preferencesRef.current.pageSize = preferences?.pageSize;
```

Comment:
1. Do you not need to clean up this if child component decides to unmount?
2. React does not guarantee that render function return value will be necessary flushed to the DOM. In [suspense](https://react.dev/reference/react/Suspense) use-case there could be fake renders which get interrupted before reaching the DOM. Only effect hooks promise reliable timing

Reply:
Changed to `useEffect` hook. I can't think of any scenario where it might be unmounted separately from the table but I don't think there's any harm in adding the clean-up.